src/Titon/Console/Tree.hh

```diff
@@ -0,0 +1,39 @@
+<?hh // strict
+/**
+ * @copyright   2010-2015, The Titon Project
+ * @license     http://opensource.org/licenses/bsd-license.php
+ * @link        http://titon.io
+ */
+
+namespace Titon\Console;
+
+/**
+ * A `Tree` object will construct the markup for a human readable of nested data.
+ *
+ * @package Titon\Console
+ */
+interface Tree {
+
+    /**
+     * Retrieve the data structure of the `Tree`.
+     *
+     * @return array<mixed>
+     */
+    public function getData(): Vector<mixed>;
```

Comment:
Use a type alias, like `TreeDataList` or something, since we're reusing it quite a bit.